scenarioo-client/app/shared/metadata/metaDataButton.directive.ts

```diff
@@ -42,11 +42,17 @@ angular.module('scenarioo.directives').directive('scMetaDataButton', ($window, L
                 scope.toggleShowingMetadata();
             });
         },
-        controller($scope) {
+        controller($scope, $route) {
             initMetadataVisibleFromLocalStorage($scope, $scope.localStorageKey);
             $scope.toggleShowingMetadata = () => {
                 $scope.linkingVariable = !$scope.linkingVariable;
                 LocalStorageService.set(STEP_METADATA_VISIBLE + $scope.localStorageKey, '' + $scope.linkingVariable);
+                // reload the page to ensure that highlighted changes are properly aligned
+                // if comparison highlights are activated and we are on the comparisons tab.
+                if ($route.current.scope.comparisonViewOptions && $route.current.scope.isComparisonChangesHighlighted()
+                    && sessionStorage.getItem('activeTab') === '2') {
```

Comment:
Should a directive trigger a page reload?

Reply:
Good point! I moved the check and the reload to step controller.
However, we do need the page reload to solve issue 1: " when toggeling details to be displayed or hidden while on the comparison view the diff highlights are not properly aligned to the underlying screen (reload of page fixes it)"